pkg/sql/schemachanger/scplan/internal/rules/helpers.go

```diff
@@ -140,6 +140,31 @@ func ColumnInSwappedInPrimaryIndex(
 	return columnInSwappedInPrimaryIndexUntyped(indexColumn.El, index.El, relationIDVar, columnIDVar, indexIDVar)
 }
 
+// IsPotentialSecondaryIndexSwap determines if a secondary index recreate is
+// occurring because of a primary key alter.
+func IsPotentialSecondaryIndexSwap(indexIdVar rel.Var, tableIDVar rel.Var) rel.Clauses {
```

Comment:
The `.` import of the `rules` package really threw me for a loop here. It might be worth considering removing in the future so changes like this don't seem broken? 😅

Reply:
We did that to reduce verbosity in the rules ecosystem. When we split them into versions originally, we did try multiple things with helpers (i.e. each rule version has its own helper or helpers use full package names to access). We eventually settled on this.